strings-csharp/DeterminingIfStringIsNumericOnly/DeterminingIfStringIsNumericOnly/IsStringNumber.cs

```diff
@@ -0,0 +1,49 @@
+using System.Text.RegularExpressions;
+using BenchmarkDotNet.Attributes;
+
+namespace DeterminingIfStringIsNumericOnly;
+
+public partial class IsStringNumber
+{
+    [GeneratedRegex("^\\d+$")]
+    private static partial Regex IsDigitRegex();
+    
+    [Benchmark]
+    public bool IsNumericWithNewRegex() => new Regex("^\\d+$").IsMatch("123456789123456789");
+
+    [Benchmark]
+    public bool IsNumericWithCompiledRegex() => IsDigitRegex().IsMatch("123456789123456789");
+
+    [Benchmark]
+    public bool IsNumericWithTryParse() => int.TryParse("123456789123456789", out _);
```

Comment:
Note that this method will return false when value is greater than int.MaxValue. For example, this string, which is clearly an integer and contains only digits, will return false "21474836471"

Reply:
You are right, thanks for pointing it out, I added a remark about this to the article and written unit tests to verify this